lsプログラムを作成

naoki00413 opened the pull request.

naoki00413 committed:
first commit

naoki00413 committed:
不要なファイルを削除しました

yoshitsugu commented on 04.ls/ls.rb:
```diff
@@ -0,0 +1,20 @@
+def option_process
+  optparse = ARGV.getopts('', 'ls:')
+  puts optparse
+end
```
これは今回使ってなさそうですかね？不要なら一旦削除でよいかと思います。

yoshitsugu commented on 04.ls/ls.rb:
```diff
@@ -0,0 +1,20 @@
+
+def file_data_list
+  lsfile_date = Dir.glob('*')
+  filename = lsfile_date.each_slice(2).to_a
+  file_sort = filename[0].zip(*filename[1..])
```
これだと必須要件の `横に最大3列を維持して表示する ` が満たせていないですね。ファイルの数が6つまでならたまたまうまくいきそうですが、それ以上だと無理かと思います。

yoshitsugu commented on 04.ls/ls.rb:
```diff
@@ -0,0 +1,20 @@
+  filename = lsfile_date.each_slice(2).to_a
+  file_sort = filename[0].zip(*filename[1..])
+  file_sort.each do |linefeed|
+    linefeed.each { print _1.ljust(10) }
```
10文字以上のファイル名があったときにくっついて表示されるので、長いファイル名でもくっつかないようにできるといいなと思います。

yoshitsugu reviewed (commented):
コメントしました！

naoki00413 committed:
指摘点を修正

ljustの数値を10から20に変更 最大３列表示できるように変更 使ってないメソッドの削除

yoshitsugu commented on 04.ls/ls.rb:
```diff
@@ -0,0 +1,15 @@
+
+def file_data_list
+  lsfile_date = Dir.glob('*')
+  filename = lsfile_date.each_slice(3).to_a
```
必須要件は横が3列なので、これだと満たせてないですね。
![Screenshot from 2023-03-13 14-29-05](https://user-images.githubusercontent.com/748445/224615623-5818ae84-a08c-4028-90a4-c6c1d5f6cb7c.png)

yoshitsugu commented on 04.ls/ls.rb:
```diff
@@ -0,0 +1,15 @@
+  filename = lsfile_date.each_slice(3).to_a
+  file_sort = filename[0].zip(*filename[1..])
+  file_sort.each do |linefeed|
+    linefeed.each { print _1.ljust(20) }
```
10から20に増やしたところで今度は20文字以上のファイル名だとダメですね。
場当たり的な対応ではなく、本質的にどういう問題なのか？を考えて対応おねがいします。また対応後は確かに解消したことを確認おねがいします。

yoshitsugu reviewed (commented):
コメントしました。

naoki00413 committed:
Merge branch 'bowling' into ls

naoki00413 committed:
達成できてなかった必須要件２つに対応するようにしました

1,横に最大３列を維持して表示するに対応しました
それに伴い、each_sliceに条件を追加しました
2, 長いファイルがくっつかないようにしました
ファイルの最大文字数を取得、そこから＋３の空白が付くようにしました
